.circleci/config.yml

```diff
@@ -291,6 +298,20 @@ commands:
           name: Wait for local UI server to start
           command: dockerize -wait tcp://localhost:4200 -timeout 4m
 
+  skip_e2e_commit:
+    description: "Halt job if commit message contains 'SKIP E2E'"
```

Comment:
would specify somewhere here that this is case insensitive. The bash `,,` is nice but also pretty obscure, I had to look this up.

Reply:
Added a comment to explain what the double comma is called.

Reply:
Thinking out loud: could we have a generic filter like `skip <task_name>`?

Reply:
Possible. One way I know how to do that is make a common command to store the commit message then each job will parse and pattern match the message.
e.g.
```

  get_commit_message:
    steps:
      - run:
          command: |
            echo 'export COMMIT_MESSAGE=$(git log -1 --pretty=format:"%s")' >> $BASH_ENV
            source $BASH_ENV
          
  e2e_presubmit_check:
    steps:
      - get_commit_message
      - run:
          - command: |
              if [[ "${COMMIT_MESSAGE,,}" == *"skip e2e"* ]]; then
                echo "Halting e2e test."
                circleci step halt
              fi
```